src/ima.js

```diff
@@ -1362,6 +1362,38 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     }
     this._textTracksHidden = [];
   }
+
+  /**
+   * When playing on same video tag need to keep the text track
+   * audio track and playback rate
+   * @private
+   * @returns {void}
+   * @instance
+   * @memberof Ima
+   */
+  _maybeSaveTracksAndRate(): void {
```

Comment:
Maybe a good name can be _maybeSavePlayerSnapshot as we call it also in other places, like the cast SDK.
And on the ad ended we can also extract it and call it _maybeRestorePlayerSnapshot

Reply:
I think the terminology of _maybeSavePlayerSnapshot should contain the current time and probably will contain the all ad break end logic as I understand the name :)